### PR TITLE
updated /create-account endpoint  and added /update-email to make it compatible with FE calls

### PR DESCRIPTION
**Pull request by sureshhmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+class HelpToSaveController @Inject() (val enrolmentStore:          EnrolmentStore,
+                                      val itmpConnector:           ITMPEnrolmentConnector,
+                                      proxyConnector:              HelpToSaveProxyConnector,
+                                      userCapService:              UserCapService,
+                                      val eligibilityCheckService: EligibilityCheckService)(
+    implicit
+    transformer: LogMessageTransformer, appConfig: AppConfig)
+  extends BaseController with EligibilityBase with EnrolmentBehaviour with WithMdcExecutionContext {
+
+  def createAccount(): Action[AnyContent] = Action.async {
```
Comment (AndyHWChung) [inline app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: For these endpoints I think there is a minimum amount of authentication we can do which is to require the AuthProvider to be either `GovernmentGateway` or `PrivilegedApplication`.

Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: e.g.
```scala
authorised(AuthProviders(GovernmentGateway, PrivilegedApplication)){ ... }
```

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: I thought about this, but how to check for the auth incase of stride requests ?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: nvm, i just realised stride auth uses `PrivilegedApplication` type too , so is the same for API quests too ?

Comment (sureshhmrc) [reply app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: is it ok if i add auth to these endpoints in a upcoming PR  ?

Comment (AndyHWChung) [reply app/uk/gov/hmrc/helptosave/controllers/HelpToSaveController.scala]: yeah, stride and privileged access through the API would come under `PrivilegedApplication` - let's add the auth to your next PR then

**Comment by AndyHWChung** [comment]
LGTM

**Pull request merged**